pages/[root]/index.tsx

```diff
@@ -49,7 +50,20 @@ export async function getStaticPaths() {
 }
 
 export const getStaticProps: GetStaticProps = async ({ params }) => {
-  return { props: { root: params?.root, pool: config.pools.find((p) => p.addresses.ROOT_CONTRACT === params?.root) } }
+  if (params) {
+    const pool = config.pools.find((p) => p.addresses.ROOT_CONTRACT === params!.root)
+
+    if (pool) {
+      // Fix to force page rerender, from https://github.com/vercel/next.js/issues/9992
+      const newProps: Props = { pool, root: params.root as string, key: pool.name || '-' }
+
+      return { props: newProps }
+    }
+
+    throw new Error(`Pool ${params.root} cannot be loaded`)
+  } else {
+    throw new Error(`Params are not passed`)
+  }
```

Comment:
Just a general best practice (taken form the Go world, but useful here as well) – please validate data first in if branches and return/error right away, to keep nesting to a minimum. 

That means you'd have a simpler control flow: 

```tsx
if (!params) {
  throw new Error(`Params are not passed`)
}

const pool = config.pools.find((p) => p.addresses.ROOT_CONTRACT === params!.root)

if (!pool) {
  throw new Error(`Pool ${params.root} cannot be loaded`)
}

// Fix to force page rerender, from https://github.com/vercel/next.js/issues/9992
const newProps: Props = { pool, root: params.root as string, key: pool.name || '-' }

return { props: newProps }
```

Reply:
Great point! I'll refactor it accordingly.